src/main/java/core/basesyntax/operation/impl/PurchaseStrategyImpl.java

```
@@ -0,0 +1,16 @@
+package core.basesyntax.operation.impl;
+
+import core.basesyntax.dao.FruitDaoImpl;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.operation.OperationHandler;
+
+public class PurchaseStrategyImpl implements OperationHandler {
+
+    @Override
+    public void updateNumberOfFruit(FruitTransaction fruitTransaction) {
+        FruitDaoImpl fruitDao = new FruitDaoImpl();
+        int newFruitQuantity =
+                fruitDao.get(fruitTransaction.getFruit()) - fruitTransaction.getQuantity();
+        fruitDao.add(fruitTransaction.getFruit(), newFruitQuantity);
```

Ensure that the `newFruitQuantity` does not become negative. If the purchase quantity exceeds the available quantity, you might want to throw an exception or handle it gracefully.